File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/WithWarriorPlugin.java
package com.dumbdogdiner.warrior;

import java.util.logging.Logger;

/**
 * Utility interface for quickly accessing plugin fields.
 */
public interface WithWarriorPlugin {
	/**
	 * @return The {@link Warrior} plugin instance.
	 */
	default Warrior getPlugin() {
		return Warrior.getInstance();
	}

	/**
	 * @return The plugin {@link Logger} provided by Bukkit.
	 */
	default Logger getLogger() {
		return this.getPlugin().getLogger();
	}
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/WithWarriorPlugin.kt
package com.dumbdogdiner.warrior

import com.dumbdogdiner.warrior.api.WarriorLogger
import java.util.logging.Logger

/**
 * Utility interface for quickly accessing plugin fields.
 */
interface WithWarriorPlugin {
    /**
     * @return The [WarriorPlugin] plugin instance.
     */
    fun getPlugin(): WarriorPlugin {
        return WarriorPlugin.instance
    }

    /**
     * @return The plugin [Logger] provided by Bukkit.
     */
    fun getLogger(): WarriorLogger {
        return WarriorPlugin.pluginLogger
    }
}